src/wp-includes/media.php

```diff
@@ -5455,6 +5459,23 @@ function wp_get_loading_attr_default( $context ) {
 		return false;
 	}
 
+	// Conditionally skip lazy-loading on images before the loop.
+	if ( 'wp_get_attachment_image' === $context || 'the_post_thumbnail' === $context ) {
+		// Only apply for main query but before the loop.
+		if ( $wp_query->before_loop && is_main_query() ) {
+			// Any image before header ends should not be lazy load.
+			$is_header_loaded = did_action( 'get_header' ) && ! doing_action( 'get_header' );
+			if ( ! $is_header_loaded ) {
+				return false;
+			}
+
+			// If footer has alredy reached then it means the current template did not include any loop. Lazy-load image from this point.
+			if ( did_action( 'get_footer' ) ) {
+				return 'lazy';
+			}
```

Comment:
Some of the logic here is slightly incorrect, for example in line 5468, the check needs to be the inverse: Only if the header _was_ loaded, we should potentially return `false`, but in addition to that we should only return `false` if the footer was _not_ yet loaded. I would suggest we simplify this logic to only return `false` here conditionally. We return `lazy` at the end of the function anyway.
```suggestion
			/*
			 * Any image before the loop, but after the header has started should not be lazy-loaded,
			 * except when the footer has already started which can happen when the current template
			 * does not include any loop.
			 */
			if ( did_action( 'get_header' ) && ! did_action( 'get_footer' ) ) {
				return false;
			}
```